### ISSUE TITLE
Move security/spam settings to form settings instead of creating an extra field

### ISSUE THREAD

**Issue by SpeGal** [contao/contao#6931]
### Description

Even though the RockSolid Antispam extension is included in the core now, the user still needs to create a "security questions" field for every single form in order for the antispam settings to work. I wish the antispam was activated by default without having to create such an extra field in the form.

For the rare cases where one might not need or want the spam protection for a form, there could be a select box in the form settings that "deactivate spam protection".

Especially in installations with a lot of forms, it would be quite helpfull not to have to create an extra field. Probably the vast majority of Contao forms need the security feature anyways. So why not integrate it by default?

Thank you guys for your great work!

**Comment by m-vo** [comment]
Keep in mind, that with the current solution, there can be a fallback (the question itself), which needs to be displayed somewhere in your form and that editors might want to place at a specific position

**Comment by SpeGal** [comment]
The fallback question would of course still be part of the spam protection. It should be displayed directly bevore the submit button anyways so the user can see it right away. I don't see the need for editors to determin any other position. And if so they might be able to edit the default question via css.

I had several instances where people complaint about getting spam requests simply because creating the anti spam field had been forgotten. In everyday life it would be just easier if one did not have to create that extra field and explain to the editors yet another thing. Bots are getting more and more aggressive. I think it should be a native Contao setting, that all forms are protected by default. 

An option to deactivate it by own risk and a standard activation is preferrable to me.

**Comment by fritzmg** [comment]
We could make it so that the field is always added depending on such a setting (e.g. before the submit field, if present, as suggested by @SpeGal) if it's not already in the list of (compiled) fields. This way you could still add it manually if you need to (to add CSS classes or change the template etc.).

**Comment by SpeGal** [comment]
Sounds good to me :)

**Comment by leofeyer** [comment]
I like the idea of configuring the SPAM protection in the form settings, maybe something like this:

<img width="583" alt="" src="https://github.com/contao/contao/assets/1192057/e255ee19-e94e-4f81-a2f1-2389e66e1670">

Then we could deprecate the `FormCaptcha` class and remove it in Contao 6.

**Comment by fritzmg** [comment]
We would still need to be able to insert the widget into the form manually, if you need to adjust CSS classes or select a different template, or if you want it on a different position. Thus I don't think we can retire the field class entirely.

**Comment by aschempp** [comment]
> We would still need to be able to insert the widget into the form manually, if you need to adjust CSS classes or select a different template, or if you want it on a different position. Thus I don't think we can retire the field class entirely.

I fully agree. It makes more sense to me to have it as a field, because it is a field. However, I agree that I always forget about adding it too … maybe we could add a hint to the form?

**Comment by leofeyer** [comment]
> if you need to adjust CSS classes or select a different template or if you want it on a different position

Although in 99% of the cases, people use the default template and always insert the CAPTCHA right before the submit button. Therefore I‘d argue that this is what the core should provide and if you need a specialized solution, use an extension that provides a CAPTCHA field or create your own.

**Comment by fritzmg** [comment]
Depending on the form and theme you will want to insert grid CSS classes for the widget for example. Having to create your own captcha field just for that purpose seems unnecessary. The captcha needs to stay a proper widget in my opinion - but can be auto-inserted as described above. I see no advantage in it not being a dedicated field.

**Comment by leofeyer** [comment]
As discussed in the Contao call, we want to add the form setting in addition to the existing CAPTCHA widget. If there is no CAPTCHA widget in the form fields, the configured CAPTCHA should be injected before the submit button.
